pkg/controller/core/workload_controller.go

```diff
@@ -172,6 +176,9 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 	}
 	log.V(2).Info("Workload update event")
 
+	// We do not handle old workload here as it will be deleted or replaced by new one anyway.
+	handlePodOverhead(wl, r.client)
```

Comment:
This is still modifying the object coming from the event.

Notice that there is a `DeepCopy` below. You should modify that object.